src/qvimagecore.cpp

```diff
@@ -156,7 +156,9 @@ void QVImageCore::loadPixmap(const ReadData &readData, bool fromCache)
 {
     // Do this first so we can keep folder info even when loading errored files
     currentFileDetails.fileInfo = readData.fileInfo;
-    updateFolderInfo();
+    currentFileDetails.loadedIndexInFolder = currentFileDetails.folderFileInfoList.indexOf(currentFileDetails.fileInfo);
+    if (currentFileDetails.loadedIndexInFolder == -1)
+        updateFolderInfo();
 
```

Comment:
When switching between images in a folder, `goToFile` has already called `updateFolderInfo` shortly before `loadPixmap` runs, so this ends up being mostly redundant aside from updating the loaded image index. We only need to run a full update if the image wasn't found in the cached list, as may happen when opening an image from a different folder (e.g. with File > Open).